### ISSUE TITLE
Allow for creation of dynamic analysis groups based on features of a case

### ISSUE THREAD

**Issue by Caustint** [broadinstitute/seqr#3872]
**Is your feature request related to a problem? Please describe.**
The analysis team frequently needs to perform analyses on subsets of cases in a project, for example looking at AIP-prioritized variants in unsolved cases. We don't have a great way to perform analyses or run searches on subsets of cases in seqr at the moment (there are certainly ways, but most require at least some exporting of information to spreadsheets). 

**Describe the solution you'd like**
In addition to the current method of defining analysis status by listing specific cases, we'd like to be able to define analysis groups based on case features. The features that can be used to define groups could mimic the filtering options for the family list on the family page: Analysis status, Analyzed by, and Data Loaded. Just like the filtering behavior on the family page, it would be important to be able to select multiple options (eg Reviewed, currently pursuing candidates + Reviewed, no clear candidate +Partial solve+ Analysis in Progress to get the list of unsolved cases)

It would be very helpful if, once defined, these analysis groups would be dynamic so that an analysis group defined by solved cases would have cases newly tagged as solved automatically added to it. 

There will be a few of these analysis groups (ie a group of unsolved cases) that I expect will be broadly useful and it would be nice to have those shared amongst all users. I am a little worried that we may clutter the analysis group list if ALL dynamically created analysis groups are automatically shared. It might be nice to have a checkbox option to make it shared or not like we do for gene lists, but that option is not critical.

**Describe alternatives you've considered**
Another option would be to have the information and links displayed on the family page (overview stats, variant tags list, project variant search link) automatically update based on how the "Families" list is filtered (eg if I filtered the family list to unsolved cases, you would see the distribution of variant tags for solved cases only and if you clicked on the AIP tag you would get a list of the AIP variants in unsolved cases only). I think this could work but may be a little dangerous since it is easy to forget that you have the families list filtered. 

**Additional context**
Rough mockup: 
![image](https://github.com/broadinstitute/seqr/assets/67653781/29e3fdd0-4d5b-47b2-aff7-b1dd980f1cf2)

**Comment by hanars** [comment]
I think for simplicity what will make the most sense is that whenever you create these groups they are project-specific, but I can define a few that are shared across all projects. You mentioned "Unsolves" will be one we definitely want for all projects, let me know if there are any others we will want to start with

**Comment by hanars** [comment]
Screenshots for how I propose this will look, let me know if theres any changes you would like:

- <img width="278" alt="Screenshot 2024-04-11 at 4 29 26 PM" src="https://github.com/broadinstitute/seqr/assets/24598672/c10108e8-5151-4bf1-b39b-68a02c73b602">
- <img width="325" alt="Screenshot 2024-04-11 at 4 29 48 PM" src="https://github.com/broadinstitute/seqr/assets/24598672/4a07d7e3-73c4-4dd5-abd4-bd7ff759233c">
- <img width="748" alt="Screenshot 2024-04-11 at 4 29 57 PM" src="https://github.com/broadinstitute/seqr/assets/24598672/4500492b-d1d4-4448-bc5c-487712649392">

**Comment by hanars** [comment]
Functionality is now live. Waiting on @Caustint to create the groups we want shared across all projects to complete the final work for this ticket

**Comment by Caustint** [comment]
There is now an "Unsolved cases" group in https://seqr.broadinstitute.org/project/R0384_rare_genomes_project_gen/project_page that we can share across all projects. Thanks!
